hawkular-wildfly-monitor/src/main/java/org/hawkular/agent/monitor/service/MonitorService.java

```diff
@@ -819,4 +832,85 @@ private void connectToFeedCommChannel() throws Exception {
         feedComm = new FeedComm(this.httpClientBuilder, this.configuration, this.feedId, this.dmrServerInventories);
         feedComm.connect();
     }
+
+    private static final class ServerAddressProvider implements InventoryStorage {
+        private final InventoryStorage delegate;
+        private final ManagedServer server;
+
+        public ServerAddressProvider(ManagedServer server, InventoryStorage delegate) throws UnknownHostException {
+            super();
+            this.delegate = delegate;
+            this.server = server;
+        }
+
+        private InetAddress[] resolveHost() throws UnknownHostException {
+            String host = null;
+            if (server instanceof RemoteDMRManagedServer) {
+                RemoteDMRManagedServer remoteServer = (RemoteDMRManagedServer) server;
+                host = remoteServer.getHost();
+            } else if (server instanceof LocalDMRManagedServer) {
+                host = InetAddress.getLocalHost().getCanonicalHostName();
+            } else {
+                throw new IllegalStateException("Unexpected type of managed server '" + server.getClass().getName()
+                        + "'; expected '" + RemoteDMRManagedServer.class.getName() + "' or '"
+                        + LocalDMRManagedServer.class.getName() + "'. Please report this bug.");
+            }
+            MsgLogger.LOG.tracef("About to resolve IP addresses for host [%s]", host);
+            return InetAddress.getAllByName(host);
+        }
+
+        @Override
+        public void storeResourceType(ResourceType<?, ?, ?, ?> resourceType) {
+            delegate.storeResourceType(resourceType);
+        }
+
+        @Override
+        public void storeResource(Resource<?, ?, ?, ?, ?> resource) {
+            final String WF_TYPE_NAME = "WildFly Server";
+            final String IP_ADDRESSES_PROPERTY_NAME = "IP Addresses";
+            String typeName = resource.getResourceType().getName().getNameString();
+            if (WF_TYPE_NAME.equals(typeName) && resource instanceof DMRResource) {
```

Comment:
> I guess in the case of extension, that IP address would still come from the base type, so we should be good here?

I understand all words but I have no idea what you are asking. What do you mean with "in the case of extension" and "base type"?